src/Compilers/Core/Portable/Operations/Operation.Enumerable.cs

```diff
@@ -0,0 +1,102 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections;
+using System;
+using System.Diagnostics;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal abstract partial class Operation : IOperation
+    {
+        internal readonly struct Enumerable : IEnumerable<IOperation>
+        {
+            private readonly Operation? _operation;
+
+            internal Enumerable(Operation operation)
+            {
+                _operation = operation;
+            }
+
+            public Enumerator GetEnumerator() => new Enumerator(_operation);
+
+            public ImmutableArray<IOperation> ToImmutableArray()
+            {
+                switch (_operation)
+                {
+                    case NoneOperation { Children: var children }:
+                        return children;
+                    case InvalidOperation { Children: var children }:
+                        return children;
+                    case null:
+                    case { } when !GetEnumerator().MoveNext():
+                        return ImmutableArray<IOperation>.Empty;
+                    default:
+                        var builder = ArrayBuilder<IOperation>.GetInstance();
+                        foreach (var child in this)
+                        {
+                            builder.Add(child);
+                        }
+                        return builder.ToImmutableAndFree();
+                }
+            }
+
+            IEnumerator<IOperation> IEnumerable<IOperation>.GetEnumerator() => this.GetEnumerator();
+            IEnumerator IEnumerable.GetEnumerator() => this.GetEnumerator();
+        }
+
+        internal struct Enumerator : IEnumerator<IOperation>
```

Comment:
📝 IIRC, you can remove the explicit implementation of `IEnumerator<IOperation>` to ensure this type does not get boxed during usage

Reply:
I don't see how I could. It still needs to implement the interface, and if it's being used as `IEnumerator`, it's already been boxed anyway.